enhancements/node/cri-o-operator.md

```diff
@@ -0,0 +1,307 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall goal of creating this new operator in OpenShift is the encapsulation
+of CRI-O related configurations, metrics and alerts within a single domain and
+namespace. This provides on one hand a higher flexibility for the Node team,
+while on the other hand the dependency on external teams gets reduced. In the
+long-term the Node team would only rely on OpenShift internal APIs to manage the
+container runtime configurations. Additional deployments for proxying runtime
+metrics are part of this operator, too.
+
+## Motivation
+
+Most of the CRI-O related configurations are today part of the [Machine Config
+Operator (MCO)][mco]. Because the MCO got bigger over time, their long-term goal
+is to provide an API for machine related configuration purposes. The new CRI-O
+operator would use this API in a later migration step.
+
+The CRI-O metrics are today served by using a [`kubelet`
+ServiceMonitor][kubelet-monitor] and a corresponding Prometheus relabeling. This
+happens as part of the [Cluster Monitoring Operator (CMO)][cmo] project. The CMO
+team also aims to provide an API for external teams to integrate their
+metrics retrieval into the OpenShift cluster monitoring. In the same way as for
+the MCO, the new CRI-O operator would use this API to integrate the runtime
+metrics retrieval natively into OpenShift by using a dedicated ServiceMonitor
+secured by TLS and RBAC.
+
+[mco]: https://github.com/openshift/machine-config-operator
+[cmo]: https://github.com/openshift/cluster-monitoring-operator
+[kubelet-monitor]: https://github.com/openshift/cluster-monitoring-operator/blob/8ec92e1/assets/control-plane/service-monitor-kubelet.yaml#L95-L116
+
+### Goals
+
+- Create a new Node owned project to be prepared for future changes of the MCO
+  and CMO.
+- Build in CRI-O metrics retrieval by using TLS and RBAC with the help of a
+  kube-rbac-proxy DaemonSet.
+- Include TLS certificate rotation (on disk as well as in a secret) for the
+  CRI-O metrics.
+
+### Non-Goals
+
+Anything which is already part of the MCO or CMO. For example, the CRI-O
+configuration management will be moved into the new operator if the API is
+available within the MCO.
+
+## Proposal
+
+The main proposal is integrating a new operator into OpenShift, which is owned
+by the Node team. This includes the whole supply chain around CI/CD and release
+management.
+
+The operator will consist of:
+
+- A dedicated namespace where the operator lives and acts on.
+- A controller managing the reconciliation of the statically provided assets,
+  which are:
+  - RBAC rules as well as a ServiceAccount for the kube-rbac-proxy based metrics
+    server.
+  - SecurityContextConstraints to allow HostPorts and HostNetwork for the
+    DaemonSet.
+  - A DaemonSet to retrieve the metrics for every node.
```

Comment:
is this referring to a world where cri-o is the single source of truth for metrics (and not cadvisor embedded in kubelet)?

Reply:
or another way to frame this, what prerequisite work is needed upstream before inclusion of this operator makes sense?

Reply:
AFAIU this is cri-o specific metrics (image pull time is a good example), not the metrics coming from cAdvisor as part of https://github.com/kubernetes/enhancements/pull/2364

Reply:
i.e. there should be no requirements on upstream kube, just cri-o

Reply:
Yes, in the first place we only refer to the CRI-O metrics.